sbi/analysis/plot.py

```diff
@@ -554,42 +554,66 @@ def handle_nan_infs(samples: List[np.ndarray]) -> List[np.ndarray]:
     return samples
 
 
+def convert_to_list_of_numpy(
+    arr: Union[List[np.ndarray], List[torch.Tensor], np.ndarray, torch.Tensor],
+) -> List[np.ndarray]:
+    """Converts a list of torch.Tensor to a list of np.ndarray."""
+    if not isinstance(arr, list):
+        arr = ensure_numpy(arr)
+        return [arr]
+    return [ensure_numpy(a) for a in arr]
+
+
+def infer_limits(
+    samples: List[np.ndarray],
+    dim: int,
+    points: Optional[List[np.ndarray]] = None,
+    eps: float = 0.1,
+) -> List:
+    """Infer limits for the plot.
+
+    Args:
+        samples: List of samples.
+        dim: Dimension of the samples.
+        points: List of points.
+        eps: Relative margin for the limits.
+    """
+    limits = []
+    for d in range(dim):
+        min_val = min(np.min(sample[:, d]) for sample in samples)
+        max_val = max(np.max(sample[:, d]) for sample in samples)
+        if points is not None:
+            min_val = min(min_val, min(np.min(point[:, d]) for point in points))
+            max_val = max(max_val, max(np.max(point[:, d]) for point in points))
+        limits.append([min_val * (1 + eps), max_val * (1 + eps)])
```

Comment:
Sorry, but I think the epsilon should be different. The current one will not work for, e.g., limits=[0, 1]. In addition, the epsilon range will not be equally big on the left and on the right side.

[Aside, I think there is a bug in the lower bound: shouldn't it be `min_val * (1-eps)`?]

I would prefer to compute the bounds as follows:
```python
max_min_range = max_val - min_val
epsilon_range = epsilon * max_min_range
limits.append([min_val - epsilon_range, max_val + epsilon_range])
```

Reply:
I think there is no bug because I am multiplying with epsilon, not adding it. 

But I agree that it is not ideal that lower and upper bounds might be scaled differently. 

Your solution looks good, I will change it.

Reply:
Great, thanks!